src/Compilers/CSharp/Test/Semantic/SourceGeneration/GeneratorDriverTests_Attributes_FullyQualifiedName.cs

```diff
@@ -569,6 +569,274 @@ internal partial void M() { }
+            step => Assert.True(step.Outputs.Single().Value is ConstructorDeclarationSyntax { Identifier.ValueText: "C" }));
+    }
+
+    [Fact]
+    public void FindConstructorAttribute2()
+    {
+        var source = """
+            using System;
+
+            partial class C
+            {
+                [CLSCompliant(true)]
+                public C()
+                {
+                }
+            }
+            """;
+        var parseOptions = TestOptions.RegularPreview;
+        Compilation compilation = CreateCompilation(source, options: TestOptions.DebugDllThrowing, parseOptions: parseOptions);
+
+        Assert.Single(compilation.SyntaxTrees);
+
+        var generator = new IncrementalGeneratorWrapper(new PipelineCallbackGenerator(ctx =>
+        {
+            var input = ctx.ForAttributeWithMetadataName<ConstructorDeclarationSyntax>("System.CLSCompliantAttribute");
+            ctx.RegisterSourceOutput(input, (spc, node) => { });
+        }));
+
+        GeneratorDriver driver = CSharpGeneratorDriver.Create(new ISourceGenerator[] { generator }, parseOptions: parseOptions, driverOptions: TestOptions.GeneratorDriverOptions);
+        driver = driver.RunGenerators(compilation);
+        var runResult = driver.GetRunResult().Results[0];
+
+        Assert.Collection(runResult.TrackedSteps["result_ForAttributeWithMetadataName"],
+            step => Assert.True(step.Outputs.Single().Value is ConstructorDeclarationSyntax { Identifier.ValueText: "C" }));
+    }
+
+    [Fact]
+    public void FindConstructorAttribute3()
+    {
+        var source = """
+            using System;
+
+            [method: CLSCompliant(true)]
+            class C()
+            {
+            }
+            """;
+        var parseOptions = TestOptions.RegularPreview;
+        Compilation compilation = CreateCompilation(source, options: TestOptions.DebugDllThrowing, parseOptions: parseOptions);
+
+        Assert.Single(compilation.SyntaxTrees);
+
+        var generator = new IncrementalGeneratorWrapper(new PipelineCallbackGenerator(ctx =>
+        {
+            var input = ctx.ForAttributeWithMetadataName<ClassDeclarationSyntax>("System.CLSCompliantAttribute");
+            ctx.RegisterSourceOutput(input, (spc, node) => { });
+        }));
+
+        GeneratorDriver driver = CSharpGeneratorDriver.Create(new ISourceGenerator[] { generator }, parseOptions: parseOptions, driverOptions: TestOptions.GeneratorDriverOptions);
+        driver = driver.RunGenerators(compilation);
+        var runResult = driver.GetRunResult().Results[0];
+
+        Assert.Collection(runResult.TrackedSteps["result_ForAttributeWithMetadataName"],
+            step => Assert.True(step.Outputs.Single().Value is ClassDeclarationSyntax { Identifier.ValueText: "C" }));
+    }
+
+    [Fact]
+    public void FindConstructorAttribute4()
+    {
+        var source = """
+            using System;
+
+            [method: CLSCompliant(true)]
+            partial class C()
```

Comment:
We need tests with multiple partial parts, preferably also with different attributes spread across the parts. Even if there are errors, we should assert the behavior.

Reply:
This feature is specifically adding support for `[method: MyGenerator]` on a primary constructor, and primary constructors can't be partial, so I'm not sure what else we're trying to test.

Could you give more detail? A concrete example of the source you'd like to see tests for would also be helpful.

Reply:
For example:

```cs
[method: CLSCompliant(true)]
partial class C() {}

[method: SomeOtherAttribute]
partial class C {}
```
or
```cs
[method: CLSCompliant(true)]
partial class C() {}

[method: SomeOtherAttribute]
partial class C() {}
```

Reply:
```csharp
[method: CLSCompliant(true)]
partial class C() {}

[method: SomeOtherAttribute]
partial class C() {}
```

This won't compile (CS8863 Only a single partial type declaration may have a parameter list). So we can't test any behavior around FAWMN there.

Reply:
> This won't compile (CS8863 Only a single partial type declaration may have a parameter list). So we can't test any behavior around FAWMN there.

Yes we can. 99% of the code source generators see is bad. We should be testing what FAWMN will do in these cases.